tardis/io/parsers/arepo.py

```diff
@@ -744,6 +741,219 @@ def create_profile(
         return self
 
 
+class CartesianProfile:
+    """
+    Class for 3D Cartesian profiles extracted from the full snapshot,
+    Only works if arepo-snap-utils are available.
+    """
+
+    def __init__(self, arepo_snapshot):
+        """
+        Parameters
+        ----------
+        arepo_snapshot : ArepoSnapshot
+            Arepo snapshot which will be mapped to 3D Grid
+
+        """
+
+        try:
+            import gadget_snap
+        except ModuleNotFoundError:
+            raise ImportError(
+                "Please make sure you have arepo-snap-util installed if you want to use 3DProfile."
+            )
+
+        self.arepo_snapshot = arepo_snapshot
+
+    def create_profile(self, res, box=None, numthreads=1):
+        """
+        Creates a profile from the full snapshot. Maps all values to
+        a 3D Cartesian grid.
+
+        Parameters
+        ----------
+        res : int
+            Resolution of the Cartesian grid
+        box : list of float
+            Region of snapshot which will be mapped to 3D.
+            Default: None
+        numthreads : int
+            Number of threads used in mapping procedure.
+
+        Returns
+        -------
+        profile : 3DProfile object
+
+        """
+
+        print("Mapping velocities to 3D grid...")
```

Comment:
Do we expect this to take long enough that print statements are necessary to track progress?

Reply:
Yes, depending on the chosen resolution, the mapping procedure can take >10 minutes (I guess if you try to map a snapshot with 55 species to a 200 grid over an hour is still realistic)